core/src/main/java/pl/agh/edu/engine/attraction/AttractionHandler.java

```diff
@@ -0,0 +1,143 @@
+package pl.agh.edu.engine.attraction;
+
+import static java.math.BigDecimal.ZERO;
+import static java.time.temporal.ChronoUnit.DAYS;
+import static pl.agh.edu.engine.attraction.AttractionState.ACTIVE;
+import static pl.agh.edu.engine.attraction.AttractionState.CHANGING_SIZE;
+import static pl.agh.edu.engine.attraction.AttractionState.INACTIVE;
+import static pl.agh.edu.engine.attraction.AttractionState.OPENING;
+import static pl.agh.edu.engine.attraction.AttractionState.SHUTTING_DOWN;
+
+import java.math.BigDecimal;
+import java.util.EnumMap;
+
+import pl.agh.edu.data.loader.JSONAttractionDataLoader;
+import pl.agh.edu.engine.bank.BankAccountHandler;
+import pl.agh.edu.engine.client.ClientGroup;
+import pl.agh.edu.engine.client.ClientGroupModifierSupplier;
+import pl.agh.edu.engine.hotel.HotelVisitPurpose;
+import pl.agh.edu.engine.room.RoomManager;
+import pl.agh.edu.engine.time.Time;
+import pl.agh.edu.engine.time.TimeCommandExecutor;
+import pl.agh.edu.engine.time.command.TimeCommand;
+import pl.agh.edu.utils.Pair;
+import pl.agh.edu.utils.RandomUtils;
+
+public class AttractionHandler extends ClientGroupModifierSupplier {
+	private final EnumMap<AttractionType, Attraction> attractions = new EnumMap<>(AttractionType.class);
+	private final BankAccountHandler accountHandler;
+	private final RoomManager roomManager;
+	private final Time time = Time.getInstance();
+	private final TimeCommandExecutor timeCommandExecutor = TimeCommandExecutor.getInstance();
+
+	public AttractionHandler(BankAccountHandler accountHandler, RoomManager roomManager) {
+		this.accountHandler = accountHandler;
+		this.roomManager = roomManager;
+	}
+
+	private boolean hasAttraction(AttractionType type) {
+		return attractions.containsKey(type);
+	}
+
+	public boolean canBuild(AttractionType type, AttractionSize size) {
```

Comment:
Tu można w jednej linijce return `return !hasAttraction(type) && accountHandler.hasOperationAbility(JSONAttractionDataLoader.buildCost.get(size));`

Reply:
wg mnie straci to na czytelności znacznie